File: app/src/main/java/com/scatl/uestcbbs/module/post/adapter/AttachmentAdapter.java
package com.scatl.uestcbbs.module.post.adapter;

import androidx.annotation.NonNull;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.AttachmentBean;

/**
 * author: sca_tl
 * date: 2020/6/25 16:14
 * description:
 */
public class AttachmentAdapter extends BaseQuickAdapter<AttachmentBean, BaseViewHolder> {
    public AttachmentAdapter(int layoutResId) {
        super(layoutResId);
    }

    public void delete(int position) {
        getData().remove(position);
        notifyItemRemoved(position);
    }

    @Override
    protected void convert(@NonNull BaseViewHolder helper, AttachmentBean item) {
        helper.setText(R.id.item_attachment_file_name, item.fileName)
                .addOnClickListener(R.id.item_attachment_delete_file);
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/post/adapter/AttachmentAdapter.kt
package com.scatl.uestcbbs.module.post.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemAttachmentBinding
import com.scatl.uestcbbs.entity.AttachmentBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder

/**
 * created by sca_tl at 2023/6/10 11:16
 */
class AttachmentAdapter: PreloadAdapter<AttachmentBean, ItemAttachmentBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemAttachmentBinding {
        return ItemAttachmentBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemAttachmentBinding>, position: Int, item: AttachmentBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.itemAttachmentFileName.text = item.fileName
    }
}